src/components/tl/cuda/alltoallv/alltoallv_ce.c

```diff
@@ -0,0 +1,405 @@
+                               UCC_EE_CUDA_STREAM);
+exit:
+    return status;
+}
+
+void ucc_tl_cuda_alltoallv_ce_progress(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_cuda_task_t *task = ucc_derived_of(coll_task, ucc_tl_cuda_task_t);
+    ucc_tl_cuda_team_t *team = TASK_TEAM(task);
+    ucc_status_t        status;
+    int                 i;
+
+    switch (task->alltoallv_ce.stage) {
+    case ALLTOALL_CE_STAGE_SYNC:
+        if (ucc_tl_cuda_get_sync(task) != UCC_OK) {
+            task->super.super.status = UCC_INPROGRESS;
+            return;
+        }
+        status = ucc_tl_cuda_alltoallv_setup_start(task);
+        if (status != UCC_OK) {
+            task->super.super.status = status;
+            return;
+        }
+        task->alltoallv_ce.stage = ALLTOALL_CE_STAGE_SETUP;
+    case ALLTOALL_CE_STAGE_SETUP:
+        status = ucc_tl_cuda_alltoallv_setup_test(task);
+        if (status != UCC_OK) {
+            task->super.super.status = status;
+            return;
+        }
+    case ALLTOALL_CE_STAGE_POST_COPIES:
+        status = ucc_tl_cuda_alltoallv_ce_post_copies(task);
+        if (ucc_unlikely(status != UCC_OK)) {
+            task->super.super.status = status;
+            return;
+        }
+        task->alltoallv_ce.stage = ALLTOALL_CE_STAGE_COPY;
+    case ALLTOALL_CE_STAGE_COPY:
+        for (i = 0; i < task->alltoallv_ce.num_posted; i++) {
+            status = ucc_ee_executor_task_test(task->alltoallv_ce.exec_task[i]);
+            if (status != UCC_OK) {
+                if (status == UCC_OPERATION_INITIALIZED) {
+                    status = UCC_INPROGRESS;
+                }
+                task->super.super.status = status;
+                return;
+            }
+        }
+        status =
+            ucc_tl_cuda_shm_barrier_start(UCC_TL_TEAM_RANK(team), task->bar);
+        if (ucc_unlikely(status != UCC_OK)) {
+            task->super.super.status = status;
+            return;
+        }
+        task->alltoallv_ce.stage = ALLTOALL_CE_STAGE_BAR;
+    default:
+        ucc_assert(task->alltoallv_ce.stage == ALLTOALL_CE_STAGE_BAR);
+        break;
+    }
+    task->super.super.status =
+        ucc_tl_cuda_shm_barrier_test(UCC_TL_TEAM_RANK(team), task->bar);
+    if (task->super.super.status == UCC_OK) {
+        ucc_tl_cuda_put_sync(task);
+    }
+}
+
+ucc_status_t ucc_tl_cuda_alltoallv_ce_start(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_cuda_task_t *task = ucc_derived_of(coll_task, ucc_tl_cuda_task_t);
+    ucc_tl_cuda_team_t *team = TASK_TEAM(task);
+
+    if (task->alltoallv_ce.stage != ALLTOALL_CE_STAGE_POST_COPIES) {
+        task->alltoallv_ce.stage = ALLTOALL_CE_STAGE_SYNC;
+    }
+
+    ucc_tl_cuda_alltoallv_ce_progress(coll_task);
+    if (task->super.super.status == UCC_INPROGRESS) {
+        ucc_progress_enqueue(UCC_TL_CORE_CTX(team)->pq, &task->super);
+        return UCC_OK;
+    }
+    return ucc_task_complete(coll_task);
```

Comment:
pls replace with, 
```suggestion
return ucc_progress_queue_enqueue(UCC_TL_CORE_CTX(team)->pq, &task->super);
```
ucc_progress_queue_enqueue will do all the necessary setup for task,